preprocessing.py

```diff
@@ -123,12 +124,18 @@ def read_tf_records(batch_size, tf_records, num_repeats=1,
     #   moving to the next file
     # The idea is to shuffle both the order of the files being read,
     # and the examples being read from the files.
-    dataset = record_list.interleave(lambda x:
-                                     tf.data.TFRecordDataset(
-                                         x, compression_type='ZLIB'),
-                                     cycle_length=64, block_length=16)
-    dataset = dataset.filter(lambda x: tf.less(
-        tf.random_uniform([1]), filter_amount)[0])
+    if sloppy_interleave:
+        dataset = record_list.apply(tf.contrib.data.parallel_interleave(
+            functools.partial(tf.data.TFRecordDataset, buffer_size=8*1024*1024, compression_type='ZLIB'),
```

Comment:
how does this perform when using a golden chunk?

Reply:
That's TBD. The tpu resnet demo has like 64 parallel reads going on in order to satisfy the TPU's demands. OTOH, our stuff compresses very well, so 1 parallel read with the same bandwidth as each of the 64 parallel reads might end up decompressing to enough MB/s of data to satisfy the cloud TPU. We'll have to see

Reply:
we've also got a PR out to provide ZLIB compression options -- should speed up the throughput substantially.